### PR TITLE
Fix the fetching credential bug. Implemented the auto-comment feature…

### PR DESCRIPTION
**Pull request by Suha1995**
… to some commands and added the unit test caes for 3 commands

## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
A few sentences describing the overall goals of the pull request's commits.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [x] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/DelineaSS/Integrations/DelineaSS/DelineaSS.py [hunk 1]
```diff
@@ -47,8 +48,8 @@ def getUsernameById(self, secret_id: str) -> str:
         url_suffix = "/api/v1/secrets/" + str(secret_id) + "/fields/username"
         return self._http_request("GET", url_suffix)
 
-    def getSecret(self, secret_id: str) -> str:
-        url_suffix = "/api/v1/secrets/" + str(secret_id)
+    def getSecret(self, secret_id: str, autocomment: str = '') -> str:
+        url_suffix = "/api/v1/secrets/" + str(secret_id) + "?autoComment=" + str(autocomment)
```
Comment (DeanArbel) [inline Packs/DelineaSS/Integrations/DelineaSS/DelineaSS.py:52]: You can pass this via the `params` argument for `_http_request` - that way `autocomment` URL argument will not be passed at all when it's not set.

File: Packs/DelineaSS/Integrations/DelineaSS/DelineaSS.py [hunk 2]
```diff
@@ -78,8 +79,8 @@ def searchSecret(self, **kwargs) -> list:
         idSecret = list(map(lambda x: x.get('id'), response))
         return idSecret
 
-    def updateSecretPassword(self, secret_id: str, new_password: str) -> str:
-        url_suffix = "/api/v1/secrets/" + str(secret_id) + "/fields/password"
+    def updateSecretPassword(self, secret_id: str, new_password: str, auto_comment: str) -> str:
+        url_suffix = "/api/v1/secrets/" + str(secret_id) + "/fields/password" + "?autoComment=" + str(auto_comment)
```
Comment (DeanArbel) [inline Packs/DelineaSS/Integrations/DelineaSS/DelineaSS.py:83]: same comment as above.

File: Packs/DelineaSS/Integrations/DelineaSS/DelineaSS.py [hunk 3]
```diff
@@ -95,12 +96,13 @@ def secret_checkin(self, secret_id: str) -> str:
...
         return self._http_request("POST", url_suffix)
 
-    def secretChangePassword(self, secret_id: str, newPassword: str) -> str:
+    def secretChangePassword(self, secret_id: str, newPassword: str, autoComment: str) -> str:
         body = {
             "newPassword": newPassword
         }
 
-        return self._http_request("POST", url_suffix="/api/v1/secrets/" + str(secret_id) + "/change-password",
+        return self._http_request("POST", url_suffix="/api/v1/secrets/" + str(
```
Comment (DeanArbel) [inline Packs/DelineaSS/Integrations/DelineaSS/DelineaSS.py:104]: same comment as above.

File: Packs/DelineaSS/Integrations/DelineaSS/DelineaSS.py [hunk 4]
```diff
@@ -145,8 +147,9 @@ def secretCreate(self, name: str, secret_template_id: str, **kwargs) -> str:
 
         return self._http_request("POST", url_suffix="/api/v1/secrets", json_data=secretjson)
 
-    def secretDelete(self, id: int) -> str:
-        return self._http_request("DELETE", url_suffix="/api/v1/secrets/" + str(id))
+    def secretDelete(self, id: int, auto_comment: str) -> str:
+        return self._http_request("DELETE",
```
Comment (DeanArbel) [inline Packs/DelineaSS/Integrations/DelineaSS/DelineaSS.py:151]: same comment as above.

File: Packs/DelineaSS/Integrations/DelineaSS/DelineaSS.yml [hunk 5]
```diff
@@ -91,6 +91,12 @@ script:
       name: newpassword
       required: true
       secret: true
+    - default: false
+      description: Secret comment
```
Comment (DeanArbel) [inline Packs/DelineaSS/Integrations/DelineaSS/DelineaSS.yml:95]: I think you should provide a better description. What does this field do?

File: Packs/DelineaSS/Integrations/DelineaSS/DelineaSS.yml [hunk 6]
```diff
@@ -222,6 +228,12 @@ script:
       name: secret_id
       required: true
       secret: false
+    - default: false
+      description: Secret comment
```
Comment (DeanArbel) [inline Packs/DelineaSS/Integrations/DelineaSS/DelineaSS.yml:232]: I think you should provide a better description. What does this field do?

File: Packs/DelineaSS/Integrations/DelineaSS/DelineaSS.yml [hunk 7]
```diff
@@ -696,6 +708,12 @@ script:
       name: id
       required: true
       secret: false
+    - default: false
+      description: Secret comment
```
Comment (DeanArbel) [inline Packs/DelineaSS/Integrations/DelineaSS/DelineaSS.yml:712]: I think you should provide a better description. What does this field do?

File: Packs/DelineaSS/Integrations/DelineaSS/DelineaSS.yml [hunk 8]
```diff
@@ -1086,6 +1104,12 @@ script:
       name: newpassword
       required: true
       secret: false
+    - default: false
+      description: Secret comment
```
Comment (DeanArbel) [inline Packs/DelineaSS/Integrations/DelineaSS/DelineaSS.yml:1108]: I think you should provide a better description. What does this field do?

**Review by DeanArbel — CHANGES_REQUESTED** [review]
The code looks good, though you could make some changes to make it better as I noted above.
Please consider changing the descriptions.

**Comment by merit-maita** [comment]
@Suha1995 i see you went over the changes requests @DeanArbel asked for, so to move forward with your contribution can you sign the [contribution registration ](https://forms.gle/XDfxU4E61ZwEESSMA) form please?

**Comment by edik24** [comment]
this PR closed.
This is the new one:
https://github.com/demisto/content/pull/26619

**Comment by Suha1995** [comment]
Yes, I have closed this old PR and raised a new PR "https://github.com/demisto/content/pull/26619"

**Pull request closed without merging**